File: Leetcode/11_CWMW.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    let maxnum = 0;
    let left = 0;
    let right = height.length - 1;

    while (left < right) {
        const width = right - left;
        const currentHeight = Math.min(height[left], height[right]);
        
        const square = width * currentHeight;
        maxnum = Math.max(maxnum, square);

        if (height[left] < height[right]) {
            left++;
        } else {
            right--;
        }
    }

    return maxnum;
    
};

console.log(maxArea(height = [1,8,6,2,5,4,8,3,7]))

File: Leetcode/11_CWMW.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left = 0
        right = len(height) - 1
        maxNum = 0

        while left < right:
            current = min(height[left], height[right]) * (right - left)
            maxNum = max(maxNum, current)
            if(height[left] < height[right]):
                left += 1
            else:
                right -= 1
        return maxNum
        
print(Solution().maxArea([1,8,6,2,5,4,8,3,7]))